src/rust/engine/src/lib.rs

```diff
@@ -92,8 +92,8 @@ impl RawNode {
           (RawStateTag::Empty as u8, externs::create_exception("No value")),
         Some(Ok(v)) =>
           (RawStateTag::Return as u8, v),
-        Some(Err(Failure::Throw(msg))) =>
-          (RawStateTag::Throw as u8, msg),
+        Some(Err(Failure::Throw(exc, _))) =>
```

Comment:
The problem is that python 2.7 doesn't attach the stack info to the exception. I could grab the stack and pass as a Value instead of a Buffer, and then render at render time, but I'm not sure it would be cheaper.